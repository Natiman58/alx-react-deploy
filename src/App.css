.Holberton-logo {
  width: 200px;
  height: auto;
}
h1 {
  display: inline;
  color: red;
  position: absolute;
  top: 30%;
  left: 218px;
  font-family: 'Times New Roman', Times, serif;
}
.App {
  position: relative;
}

.App-header {
  border-bottom: 3px solid red;
  position: relative;
}

.App-footer {
  border-top: 3px solid red;
  width: 100%;
  position: absolute;
  bottom: -433px;
  text-align: center;
  font-family: 'Times New Roman', Times, serif;
  font-style: italic;
}

.App-body {
  padding: 30px;
}

form {
  white-space: pre-wrap;
}
.Ok-button {
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
}