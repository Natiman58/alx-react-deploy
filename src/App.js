import logo from './holberton-logo.jpg';
import './App.css';
import  { getFullYear, getFooterCopy }  from './utils.js';
import React from 'react';


function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className='Holberton-logo' alt="Holberton logo"/>
        <h1>School dashboard</h1>
      </header>

      <div className='App-body'>
        <p>Login to access the full dashboard</p>
        <form>
        <label htmlFor='email'> Email: <input type='text' name='email'/> </label>
        <label htmlFor='password'> Password: <input type='text' name='password'/> </label>
        <button className='Ok-button' type='submit'>OK</button>
        </form>
      </div>

      <footer className='App-footer'>
        <p>Copyright { getFullYear() } { getFooterCopy("True") }</p>

      </footer>
    </div>
  );
}

export default App;
