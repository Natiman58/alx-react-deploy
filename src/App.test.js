import App from './App';
import renderer from 'react-test-renderer';


test('App is rendered without crashing', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('App renders a div with the class App-header', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // To check the presence of div with the class App-header
  const appHeader = tree.children.find((child) => child.props.className === 'App-header');
  expect(appHeader).toBeDefined();
});

test('App renders a div with the class App-body', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // To check the presence of the div with the class App-body
  const appBody = tree.children.find((child) => child.props.className === 'App-body');
  expect(appBody).toBeDefined();
});

test('App renders a div with the class App-footer', () => {
  const component = renderer.create(<App/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // To check the presence of div with the class App-footer
  const appFooter = tree.children.find((child) => child.props.className === "App-footer");
  expect(appFooter).toBeDefined();
});
