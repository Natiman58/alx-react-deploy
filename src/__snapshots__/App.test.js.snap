// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is rendered without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="Holberton logo"
      className="Holberton-logo"
      src="holberton-logo.jpg"
    />
    <h1>
      School dashboard
    </h1>
  </header>
  <div
    className="App-body"
  >
    <p>
      Login to access the full dashboard
    </p>
    <form>
      <label
        htmlFor="email"
      >
         Email: 
        <input
          name="email"
          type="text"
        />
         
      </label>
      <label
        htmlFor="password"
      >
         Password: 
        <input
          name="password"
          type="text"
        />
         
      </label>
      <button
        className="Ok-button"
        type="submit"
      >
        OK
      </button>
    </form>
  </div>
  <footer
    className="App-footer"
  >
    <p>
      Copyright 
      2023
       
      Holberton School
    </p>
  </footer>
</div>
`;

exports[`App renders a div with the class App-body 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="Holberton logo"
      className="Holberton-logo"
      src="holberton-logo.jpg"
    />
    <h1>
      School dashboard
    </h1>
  </header>
  <div
    className="App-body"
  >
    <p>
      Login to access the full dashboard
    </p>
    <form>
      <label
        htmlFor="email"
      >
         Email: 
        <input
          name="email"
          type="text"
        />
         
      </label>
      <label
        htmlFor="password"
      >
         Password: 
        <input
          name="password"
          type="text"
        />
         
      </label>
      <button
        className="Ok-button"
        type="submit"
      >
        OK
      </button>
    </form>
  </div>
  <footer
    className="App-footer"
  >
    <p>
      Copyright 
      2023
       
      Holberton School
    </p>
  </footer>
</div>
`;

exports[`App renders a div with the class App-footer 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="Holberton logo"
      className="Holberton-logo"
      src="holberton-logo.jpg"
    />
    <h1>
      School dashboard
    </h1>
  </header>
  <div
    className="App-body"
  >
    <p>
      Login to access the full dashboard
    </p>
    <form>
      <label
        htmlFor="email"
      >
         Email: 
        <input
          name="email"
          type="text"
        />
         
      </label>
      <label
        htmlFor="password"
      >
         Password: 
        <input
          name="password"
          type="text"
        />
         
      </label>
      <button
        className="Ok-button"
        type="submit"
      >
        OK
      </button>
    </form>
  </div>
  <footer
    className="App-footer"
  >
    <p>
      Copyright 
      2023
       
      Holberton School
    </p>
  </footer>
</div>
`;

exports[`App renders a div with the class App-header 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="Holberton logo"
      className="Holberton-logo"
      src="holberton-logo.jpg"
    />
    <h1>
      School dashboard
    </h1>
  </header>
  <div
    className="App-body"
  >
    <p>
      Login to access the full dashboard
    </p>
    <form>
      <label
        htmlFor="email"
      >
         Email: 
        <input
          name="email"
          type="text"
        />
         
      </label>
      <label
        htmlFor="password"
      >
         Password: 
        <input
          name="password"
          type="text"
        />
         
      </label>
      <button
        className="Ok-button"
        type="submit"
      >
        OK
      </button>
    </form>
  </div>
  <footer
    className="App-footer"
  >
    <p>
      Copyright 
      2023
       
      Holberton School
    </p>
  </footer>
</div>
`;
